File: backend/src/Organi.Server/Organi.Server.Domain/Entities/Review.cs
using Organi.Server.Domain.Common;

namespace Organi.Server.Domain.Entities;

public sealed class Review : AuditableEntity
{
    public int Rating { get; set; }
    public string? Title { get; set; }
    public string? Comment { get; set; }
    public bool IsApproved { get; set; }
    public Guid ProductId { get; set; }
    public Guid UserId { get; set; }

    public Product Product { get; set; } = default!;
    public User User { get; set; } = default!;
}

File: frontend/web/src/types/api/review.ts
// Mirrors Organi.Server.Application.Features.Reviews.DTOs.ReviewResponse (public, per-product)
export interface ReviewResponse {
  id: string
  rating: number
  title: string | null
  comment: string | null
  productId: string
  userId: string
  userFullName: string
  createdAt: string
  updatedAt: string | null
}

// Mirrors CreateReviewCommand (ProductId travels in the URL, not the body)
export interface CreateReviewRequest {
  rating: number
  title: string | null
  comment: string | null
}

// Mirrors Organi.Server.Application.Features.Reviews.DTOs.AdminReviewResponse
export interface AdminReviewResponse {
  id: string
  rating: number
  title: string | null
  comment: string | null
  productId: string
  productName: string
  userId: string
  userFullName: string
  createdAt: string
  updatedAt: string | null
}
